### PR TITLE
Feature/116 use latest efcore packages fitting to dotnet framework

### PR DESCRIPTION
**Pull request by rsr-maersk**
net6.0 to use efcore6
net5.0 to use efcore5

Remnove previews

### CODE REVIEW

**Comment by rsr-maersk** [comment]
@artiomchi or anyone. looks like we are ready for a official dotnet6 versions.
Any comments?

**Comment by rsr-maersk** [comment]
@grante @ObeA  @LonghronShen  and contributors in the house?

**Comment by generik0** [comment]
Anyone? @artiomchi 
Should be a simple PR = this one?

And happy new years

File: src/FlexLabs.EntityFrameworkCore.Upsert/FlexLabs.EntityFrameworkCore.Upsert.csproj [hunk 1]
```diff
@@ -50,8 +50,12 @@ v4.0.0
     <PostBuildEvent>signtool.exe sign /n "$(SignCertificateName)" /fd sha256 /tr "http://timestamp.digicert.com" /td sha256 FlexLabs.EntityFrameworkCore.Upsert.dll</PostBuildEvent>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.0-preview.5.21301.9" />
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="5.0.1" />
```
Comment (artiomchi) [inline src/FlexLabs.EntityFrameworkCore.Upsert/FlexLabs.EntityFrameworkCore.Upsert.csproj:54]: Ideally I'd prefer the package to only be referenced once, and have a property setting the version..

Something like 

```
  <PropertyGroup>
    <EFCoreVersion>5.0.1</EFCoreVersion>
  </PropertyGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
    <EFCoreVersion>6.0.1</EFCoreVersion>
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="$(EFCoreVersion)" />
  </ItemGroup>
```

**Comment by artiomchi** [comment]
Hey there, sorry for my absence, I don't have as much time to work on this project these days, but I'll go through the PRs and issues today/this week, as well as through this ticket specifically before the end of the day!

**Comment by generik0** [comment]
No worrie @artiomchi 
This time We did all the work for you.

There are breaking changes between the ef cores, hence, If We want to support ef 5 anf 6 in this version, We need to keep both...

**Comment by rsr-maersk** [comment]
@artiomchi changed to property condition for efcore version. I have defaulted to efcore 6 not 5 as in your comment. Net6/Ef6 is LTS, and efcore5 will be out of support sooner rather than later..

**Comment by artiomchi** [comment]
@rsr-maersk yeah, it doesn't matter either way, since we end up setting specific different versions for each framework that it was compiled for
I've pushed a couple commits to patch the builds (GitHub actions environment update), and the csproj references while I was at it

**Comment by generik0** [comment]
Cool, do we need to change the package version to v6.0.x too?

Anything else missing?
I sort of need this Monday (tomorrow;-)

**Comment by artiomchi** [comment]
@generik0 yup - I'll bump that as well before the release tonight.
Just have one last question unanswered on the PR, but it's in the test project, so won't prevent me from merging it - I just want to understand why it was done :)

**Comment by rsr-maersk** [comment]
@artiomchi it is very strange, i dont see any review comments / questions for the PR.

**Comment by artiomchi** [comment]
Did a bit of checking and figured it out - it was the timestamp handling for PostgreSQL.. I've changed to a different implementation that uses UTC datetimes stored in the database (which should be the approach in most cases anyway)

**Pull request merged**